### PR TITLE
[MB-1599] Coordinator node should be started first to start other nodes

### PR DESCRIPTION
**Pull request by sajinidesilva**
jira: https://wso2.org/jira/browse/MB-1599

### CODE REVIEW

**Comment by sajinidesilva** [comment]
Public jira : https://wso2.org/jira/browse/MB-1599

File: modules/andes-core/broker/src/main/java/org/wso2/andes/thrift/MBThriftClient.java [hunk 1]
```diff
@@ -212,24 +228,30 @@ public static synchronized void clearAllActiveSlotRelationsToQueue(String queueN
...
+                int thriftCoordinatorServerPort = Integer.parseInt(thriftCoordinatorServerPortString);
+                transport = new TSocket(thriftCoordinatorServerIP,
+                        thriftCoordinatorServerPort);
+                try {
+                    transport.open();
+                    TProtocol protocol = new TBinaryProtocol(transport);
+                    return new SlotManagementService.Client(protocol);
+                } catch (TTransportException e) {
+                    log.error("Could not initialize the Thrift client. " + e.getMessage(), e);
+                    throw new TTransportException(
```
Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/thrift/MBThriftClient.java]: this is unusual.... why do we have to wrap using a another exception?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/thrift/MBThriftClient.java [hunk 2]
```diff
@@ -372,6 +394,9 @@ public static synchronized long updateSlotDeletionSafeZone(long safeZoneMessageI
                 handleCoordinatorChanges();
                 throw new ConnectionException("Coordinator has changed", e);
             }
+        } catch (ThriftClientException e) {
+            log.error("Error occurred while receiving coordinator details from map");
```
Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/thrift/MBThriftClient.java]: shouldn't we want to log the 'e' as well?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/thrift/exception/ThriftClientException.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+public class ThriftClientException extends Exception {
+
+    public ThriftClientException() {
+    }
+
+    public ThriftClientException(String message) {
+        super(message);
+    }
+
+    public ThriftClientException(Throwable cause) {
```
Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/thrift/exception/ThriftClientException.java]: do we have a usage for passing only a throwable without a meaningful message?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/thrift/MBThriftClient.java [hunk 4]
```diff
@@ -83,6 +84,9 @@ public static synchronized Slot getSlot(String queueName,
                 throw new ConnectionException("Coordinator has changed", e);
             }
 
+        } catch (ThriftClientException e) {
+            handleCoordinatorChanges();
+            throw new ConnectionException(e.getMessage());
```
Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/thrift/MBThriftClient.java]: only the message? any particular reason?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/thrift/MBThriftClient.java (hunk 1 above)
Comment (sajinidesilva) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/thrift/MBThriftClient.java]: This will add more details to the general exception.

**Pull request merged**